system.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+try:
+    import netifaces
+except ImportError:
+    netifaces = None
+import os
+import platform
+from subprocess import Popen, PIPE
+import sys
+import time
+import psutil
+import plugins
+try:
+    import distro
+except ImportError:
+    distro = None
+
+def systemCommand(Command, newlines=True):
+    Output = ""
+    Error = ""
+    try:
+        proc = Popen(Command.split(), stdout=PIPE)
+        Output = proc.communicate()[0]
+    except Exception:
+        pass
+
+    if Output:
+        if newlines is True:
+            Stdout = Output.split("\n")
+        else:
+            Stdout = Output
+    else:
+        Stdout = []
+    if Error:
+        Stderr = Error.split("\n")
+    else:
+        Stderr = []
+
+    return (Stdout, Stderr)
+
+
+def linux_hardware_memory():
+    block_size = 0
+    try:
+        with open("/sys/devices/system/memory/block_size_bytes", "r") as f:
+            block_size = int(f.readline().strip(), 16)
+
+        memory = 0
+        with os.scandir("/sys/devices/system/memory/") as it:
+            for entry in it:
+                if not entry.name.startswith("memory"):
+                    continue
+                with open(entry.path + "/state", "r") as f:
+                    if "online" != f.readline().strip():
+                        continue
+                    else:
+                        memory += block_size
+
+        return memory
+    except Exception:
+        return 0
+
+def get_public_ip_cpnat():
+    try:
+        with open('/var/cpanel/cpnat','r') as file:
+            public_ip = file.read().split()[1]
+        return public_ip
+    except FileNotFoundError:  # In case this is not a cPanel server or not a nat server
+        output = os.popen('curl https://myip.cpanel.net 2>/dev/null').read()
```

Comment:
This hostname is only accessible by ipv4? Otherwise we have to use the `-4` flag and maybe separately try to fetch the ipv6 with the `-6` flag.